### PR TITLE
update versions of app-designer

### PR DESCRIPTION
**Pull request by iamhks**
Addresses odk-x/tool-suite-X#248

#### What is included in this PR?
Documentation on how to update versions of the app-designer software

### CODE REVIEW

File: odkx-src/build-app.rst [hunk 1]
```diff
@@ -831,3 +831,19 @@ To update any app level or table level files, or to modify the database schema (
     .. warning::
 
       Resetting the app server will start a new data set. If you want to keep the old data, you should download it to a separate database.
+
+:th:`To update versions`: 
+
+You need to download the new `app designer <https://github.com/odk-x/app-designer/releases/tag/2.1.7>`_  , and delete the unneeded default files using
```
Comment (linl33) [inline odkx-src/build-app.rst:837]: Use `https://github.com/odk-x/app-designer/releases/latest`, this link redirects to the latest tag.

File: odkx-src/build-app.rst [hunk 2]
```diff
@@ -831,3 +831,19 @@ To update any app level or table level files, or to modify the database schema (
...
+
+  $ grunt empty
+
+Then copy over your tables folders within :file:`config/tables`, and then copy over anything customized :th:`framework, csv` into the :file:`config/assets` (sub-) folders. Then re-convert everything using 
+
+.. code-block:: console
+
+  $ grunt xlsx-convert-all 
+
+and your app designer should be set. You will also need to update software on tabs and server to the same version as well.
```
Comment (linl33) [inline odkx-src/build-app.rst:849]: ~~tabs~~
Android devices

File: odkx-src/build-app.rst [hunk 3]
```diff
@@ -831,3 +831,19 @@ To update any app level or table level files, or to modify the database schema (
...
+
+:th:`To update versions`: 
+
+You need to download the new `app designer <https://github.com/odk-x/app-designer/releases/tag/2.1.7>`_  , and delete the unneeded default files using
+
+.. code-block:: console
+
+  $ grunt empty
+
+Then copy over your tables folders within :file:`config/tables`, and then copy over anything customized :th:`framework, csv` into the :file:`config/assets` (sub-) folders. Then re-convert everything using
```
Comment (linl33) [inline odkx-src/build-app.rst:843]: The entire `assets` directory should be copied over. But for the `framework`, if the user has customized it **and** ODK-X has also updated it, then the user will need to manually merge the 2 copies.

**Review by linl33 — CHANGES_REQUESTED** [review]
Thanks for documenting this.

File: odkx-src/build-app.rst (hunk 3 above)
Comment (iamhks) [reply odkx-src/build-app.rst]: @linl33 Noted all and updated.

**Review by elmps2018 — APPROVED** [review]

**Pull request merged**